pkg/apis/seedmanagement/helper/helper.go

```diff
@@ -0,0 +1,122 @@
+		return nil, err
+	}
+	return cfg, nil
+}
+
+// DecodeGardenletConfig decodes the given raw extension into an external GardenletConfiguration version.
+func DecodeGardenletConfigExternal(rawConfig *runtime.RawExtension, withDefaults bool) (*configv1alpha1.GardenletConfiguration, error) {
+	cfg, err := DecodeGardenletConfig(rawConfig, withDefaults)
+	if err != nil {
+		return nil, err
+	}
+	return ConvertGardenletConfigExternal(cfg)
+}
+
+// ConvertGardenletConfigExternal converts the given internal GardenletConfiguration version to an external version.
+func ConvertGardenletConfigExternal(cfg *config.GardenletConfiguration) (*configv1alpha1.GardenletConfiguration, error) {
+	obj, err := scheme.ConvertToVersion(cfg, configv1alpha1.SchemeGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	result, ok := obj.(*configv1alpha1.GardenletConfiguration)
+	if !ok {
+		return nil, fmt.Errorf("could not convert GardenletConfiguration to version %s", configv1alpha1.SchemeGroupVersion.String())
+	}
+	return result, nil
+}
+
+// ConvertSeed converts the given external Seed version to an internal version.
+func ConvertSeed(seed *gardencorev1beta1.Seed) (*gardencore.Seed, error) {
+	obj, err := scheme.ConvertToVersion(seed, gardencore.SchemeGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	result, ok := obj.(*gardencore.Seed)
+	if !ok {
+		return nil, fmt.Errorf("could not convert Seed to internal version")
+	}
+	return result, nil
+}
+
+// ConvertSeedExternal converts the given internal Seed version to an external version.
+func ConvertSeedExternal(seed *gardencore.Seed) (*gardencorev1beta1.Seed, error) {
+	obj, err := scheme.ConvertToVersion(seed, gardencorev1beta1.SchemeGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	result, ok := obj.(*gardencorev1beta1.Seed)
+	if !ok {
+		return nil, fmt.Errorf("could not convert Seed to version %s", gardencorev1beta1.SchemeGroupVersion.String())
+	}
+	return result, nil
+}
+
+func getDecoder(withDefaulter bool) runtime.Decoder {
+	if withDefaulter {
+		return serializer.NewCodecFactory(scheme).UniversalDecoder()
+	}
+	return versioning.NewCodec(nil, serializer.NewCodecFactory(scheme).UniversalDeserializer(), runtime.UnsafeObjectConvertor(scheme),
+		scheme, scheme, nil, runtime.DisabledGroupVersioner, runtime.InternalGroupVersioner, scheme.Name())
+
+}
+
+// Bootstrap returns the value of the Bootstrap field of the given gardenlet, or None if the field is nil.
+func Bootstrap(gardenlet *seedmanagement.Gardenlet) seedmanagement.Bootstrap {
+	if gardenlet.Bootstrap != nil {
+		return *gardenlet.Bootstrap
+	}
+	return seedmanagement.BootstrapNone
+}
+
+// MergeWithParent returns the value of the MergeWithParent field of the given gardenlet, or false if the field is nil.
+func MergeWithParent(gardenlet *seedmanagement.Gardenlet) bool {
+	if gardenlet.MergeWithParent != nil {
+		return *gardenlet.MergeWithParent
+	}
+	return false
+}
```

Comment:
Can this be replaced by using https://github.com/gardener/gardener/blob/master/pkg/utils/miscellaneous.go#L123-L126?

Reply:
Yes, changed the `Bootstrap` function to have a similar interface for consistency.